raiden/tests/utils/factories.py

```diff
@@ -327,39 +372,52 @@ def make_signed_balance_proof(
 
     signature = eth_sign(privkey=private_key, data=data_to_sign)
 
-    signed_balance_proof = BalanceProofSignedState(
-        nonce,
-        transferred_amount,
-        locked_amount,
-        locksroot,
-        token_network_address,
-        channel_identifier,
-        extra_hash,
-        signature,
-        sender_address,
-        UNIT_CHAIN_ID,
+    return BalanceProofSignedState(
+        nonce=nonce,
+        transferred_amount=transferred_amount,
+        locked_amount=locked_amount,
+        locksroot=locksroot,
+        token_network_identifier=token_network_address,
+        channel_identifier=channel_identifier,
+        message_hash=extra_hash,
+        signature=signature,
+        sender=sender_address,
+        chain_id=UNIT_CHAIN_ID,
     )
 
-    return signed_balance_proof
-
 
 def make_signed_transfer_for(
-        channel_state,
-        amount,
-        initiator,
-        target,
-        expiration,
-        secret,
-        identifier=1,
-        nonce=1,
-        transferred_amount=0,
-        locked_amount=None,
-        pkey=UNIT_TRANSFER_PKEY,
-        sender=UNIT_TRANSFER_SENDER,
-        compute_locksroot=False,
-        allow_invalid=False,
+        channel_state: NettingChannelState = EMPTY,
+        amount: typing.TokenAmount = EMPTY,
+        initiator: typing.InitiatorAddress = EMPTY,
+        target: typing.TargetAddress = EMPTY,
+        expiration: typing.BlockExpiration = EMPTY,
+        secret: typing.Secret = EMPTY,
+        identifier: typing.PaymentID = EMPTY,
+        nonce: typing.Nonce = EMPTY,
+        transferred_amount: typing.TokenAmount = EMPTY,
+        locked_amount: typing.TokenAmount = EMPTY,
+        pkey: PrivateKey = EMPTY,
+        sender: typing.Address = EMPTY,
+        compute_locksroot: typing.Locksroot = EMPTY,
+        allow_invalid: bool = EMPTY,
 ):
 
+    channel_state = if_empty(channel_state, make_channel_state())
+    amount = if_empty(amount, 0)
+    initiator = if_empty(initiator, make_address())
+    target = if_empty(target, make_address())
+    expiration = if_empty(expiration, UNIT_REVEAL_TIMEOUT)
+    secret = if_empty(secret, make_secret())
+    identifier = if_empty(identifier, 1)
+    nonce = if_empty(nonce, 1)
+    transferred_amount = if_empty(transferred_amount, 0)
+    # locked_amount = if_empty(locked_amount, EMPTY)
```

Comment:
why is this commented out? And I guess default value should not be `EMPTY` but `0`.

Reply:
otherwise tests fail, I assume this is passed to another function which expects `EMPTY` as the default argument.

Reply:
Hmm why though? It's passed right below to `make_signed_transfer()` and it's then populated [here](https://github.com/raiden-network/raiden/pull/3028/files/5e210129d92543b2fb6047646a378c5169a7592d#diff-c98d94b00ee0ca87ec14bf5ffe4f5585R294) if it's empty with the same value as the transfer amount.

Reply:
the previous version of the code used `locked_amount = locked_amount or 0`, meaning the value was always changed from `None` to `0`, and in the inner function the check was `locked_amount is None`, which would not be the case anymore:

https://github.com/raiden-network/raiden/pull/3028/files/5e210129d92543b2fb6047646a378c5169a7592d#diff-c98d94b00ee0ca87ec14bf5ffe4f5585L275

The result was a improperly initialized variable, with the value `0` instead of the value of the `lock.amount`. If I change this to zero, again, the same problem will happen and the unit tests will fail.